src/components/Home.jsx

```diff
@@ -1,19 +1,97 @@
-import React from 'react';
+import React, { useState } from 'react';
 import getToken from '../lib/tokens';
 import { useHistory } from 'react-router-dom';
+import db from '../lib/firebase';
+import firebase from 'firebase/app';
+
+import './Home.css';
 
 const Home = () => {
   let history = useHistory();
+  const [tokenName, setTokenName] = useState('');
+  const [notification, setNotification] = useState(null);
 
   const generateToken = () => {
     const token = getToken();
     localStorage.setItem('token', token);
+
+    db.collection('tokens')
+      .doc('tokenDocument')
+      .update({
+        tokenArray: firebase.firestore.FieldValue.arrayUnion(token),
+      })
+      .catch((error) => {
+        db.collection('tokens')
+          .doc('tokenDocument')
+          .set({
+            tokenArray: [token],
+          });
+      });
+
     history.push('/list-view');
   };
 
+  const resetError = () => {
+    setTimeout(() => {
+      setNotification(null);
+    }, 10000);
+  };
+
+  const compareToken = (e) => {
+    e.preventDefault();
+
+    db.collection('tokens')
+      .where('tokenArray', 'array-contains', tokenName)
+      .get()
+      .then((querySnapshot) => {
+        let tokenInDB;
+        querySnapshot.forEach((doc) => {
+          tokenInDB = doc.exists;
+        });
+        if (tokenInDB) {
+          localStorage.setItem('token', tokenName);
+          history.push('/list-view');
+        } else {
+          setNotification(
+            'Token not found - please check spelling and try again or create new list',
+          );
+        }
+        resetError();
+      })
+      .catch((error) => {
+        setNotification(
+          'An unexpected error occurred, please refresh the page and try again',
+        );
+        resetError();
+      });
+  };
+
   return (
-    <div>
+    <div className="home">
+      <h1>Welcome to your Smart Shopping list!</h1>
       <button onClick={generateToken}>Create a new list</button>
+      <p> - or - </p>
+      <p>Join an existing shopping list by entering a three word token.</p>
+
+      <form onSubmit={compareToken}>
+        <label htmlFor="inputToken" className="shareToken">
+          Share token
+          <br />
+          {notification ? (
+            <div className="errorMessage">{notification}</div>
+          ) : null}
```

Comment:
nit: I think something like this would work too.
```jsx
          {notification && (<div className="errorMessage">{notification}</div>)}
```

and then consider moving this outside of the label element. I think validation messages are typically displayed below the field.